docs/topics/whatsnew21.md

```diff
@@ -10,10 +23,1562 @@ All you need to do is [change the Kotlin version](configure-build-for-eap.md) to
+This should make multiplatform development for Apple targets easier.
+
+The current basic support includes the ability to:
+
+* Export multiple Gradle modules from Kotlin directly to Swift.
+* Define custom Swift module names with the `moduleName` property.
+* Set collapse rules for the package structure with the `flattenPackage` property.
+
+You can use the following build file in your project as a starting point for setting up Swift export:
+
+```kotlin
+// build.gradle.kts 
+kotlin {
+
+    iosX64()
+    iosArm64()
+    iosSimulatorArm64()
+
+    @OptIn(ExperimentalSwiftExportDsl::class)
+    swiftExport {
+        // Root module name
+        moduleName = "Shared"
+
+        // Collapse rule
+        // Removes package prefix from generated Swift code
+        flattenPackage = "com.example.sandbox"
+
+        // Export external modules
+        export(project(":subproject")) {
+            // Exported module name
+            moduleName = "Subproject"
+            // Collapse exported dependency rule
+            flattenPackage = "com.subproject.library"
+        }
+
+        // Configure Swift export binaries
+        binaries {
+            linkTaskProvider.configure {
+                freeCompilerArgs += "-opt-in=some.value"
+            }
+        }
+    }
+}
+```
+
+You can also clone our [public sample](https://github.com/Kotlin/swift-export-sample) with Swift export already set up.
+
+The compiler automatically generates all the necessary files (including `swiftmodule` files, 
+static `a` library, and header and `modulemap` files) and copies them into the app's build directory,
+which you can access from Xcode.
+
+#### How to enable Swift export
+
+This feature is currently [Experimental](components-stability.md#stability-levels-explained).
+To try it out in your project, add the following Gradle option to your `gradle.properties` file:
+
+```none
+# gradle.properties
+kotlin.experimental.swift-export.enabled=true
+```
+
+#### Leave feedback on Swift export
+
+We're planning to expand and stabilize Swift export support in future Kotlin releases.
+Please leave your feedback in [this YouTrack issue](https://youtrack.jetbrains.com/issue/KT-64572).
+
+### Ability to publish Kotlin libraries from any host
+
+> This feature is currently [Experimental](components-stability.md#stability-levels-explained).
+> Opt-in is required (see the details below), and you should use it only for evaluation purposes.
+> We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-71290).
+>
+{style="warning"}
+
+The Kotlin compiler produces `.klib` artifacts for publishing Kotlin libraries.
+Previously, you could get the necessary artifacts from any host, except for Apple platform targets that required a Mac machine. 
+That put a special restraint on Kotlin Multiplatform projects that targeted iOS, macOS, tvOS, and watchOS targets.
```

Comment:
```suggestion
That puts a special restraint on Kotlin Multiplatform projects that targeted iOS, macOS, tvOS, and watchOS targets.
```

Reply:
Since we used "previously", it means "in the past". No change needed.